mrf.html

```diff
@@ -0,0 +1,309 @@
+                .then( menu => {
+                    if(menu == []) {
+                        document.getElementById('messtable').innerHTML = `<tr><td colspan=7><h1 style="text-align:center;">To Be Updated</h1></td></td>`;
+                        return;
+                    }
+                    menu = JSON.parse(menu);
+                    menu.map(row => {
+                        let indiv_row = `
+                        <tr>
+                            <td>${row['time']}</td>
+                            <td>${row['monday']}</td>
+                            <td>${row['tuesday']}</td>
+                            <td>${row['wednesday']}</td>
+                            <td>${row['thursday']}</td>
+                            <td>${row['friday']}</td>
+                            <td>${row['saturday']}</td>
+                            <td>${row['sunday']}</td>
+                        </tr>
+                        `
+                        if (indiv_row != undefined)
+                        {
+                            output += indiv_row;
+                        }
+                    });
+                })
+                .then(() => {
+                    console.log(output);
+                    console.log("here");
+                    document.getElementById('messtable').innerHTML = output;
+                })
+                .catch(() => {
+                    document.getElementById('messtable').innerHTML = `<tr><td colspan=7><h1 style="text-align:center;">To Be Updated</h1></td></td>`;
+                    return;
+                });
+        }
+    
+    function getQue(){
+        var roll_no = document.getElementById("rollinput").value;
+        (async () => {
+              const rawResponse = await fetch('https://afd04240.ngrok.io/que', {
+                method: 'POST',
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({"roll_no": roll_no})
+              });
+              const content = await rawResponse.json();
+            
+            document.getElementById("question").innerHTML = content['que'];
+            })();
+    }
+    
+    function onSubmit(){
+        var roll_no = document.getElementById("rollinput").value;
+     var pwd = document.getElementById("pass").value;
+    var ans = document.getElementById("ansinput").value;
+    
+    (async () => {
+              const rawResponse = await fetch('https://afd04240.ngrok.io/check', {
+                method: 'POST',
+                headers: {
+                  'Accept': 'application/json',
+                  'Content-Type': 'application/json'
+                },
+                body: JSON.stringify({"roll_no": roll_no, "password": pwd,  "answer":ans})
+              });
+              const content = await rawResponse.json();
+
+              console.log(content);
+            })();
+        
+    }

```

Comment:
Error handling needs to be done for the following cases:

- Failed to get security question: Raise a browser alert that "Roll number is incorrect!"
- Failed to login: Raise that "The entered credentials are incorrect!"
- Failed to update: Raise that "Internal server error! Please try later..."
- Successful: Clear the form entries and raise "Your request is submitted for approval!"

Disable the entire portal on Sunday. A check on the backend needs to be made so not possible to do it on Sunday. If the user tampers with the dates,the server should return "Inactive Day!" and the same should be alerted to the user after which a page reload should be triggered.

Only activate fields for password and security answer once get security question returns a security question.

Only show the dates from the upcoming Monday to Sunday and in case the user tampers on the frontend, the backend checks and returns "Invalid Date Values" which is alerted to the user and a page reload is triggered.

By raise I mean use the JS `alert` function.

Reply:
@139Vaibhav @rakaar @PrashikD Please take note and improvise your parts accordingly. The best way to handle error alerts is to check if response was not successful and alery whatever message backend sent. This way it can be totally managed on the backend side. @PrashikD and @rakaar cooperate on what messages and status codes you'll be sending so that accoridngly reload or clearing of form can be triggered.